modules/common/secret/types.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2024 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secret
+
+// PasswordSelector
+type PasswordSelector struct {
+	// +kubebuilder:validation:Optional
+	// Service - Selector to get the service password from the Secret
+	Service string `json:"service"`
+	// +kubebuilder:validation:Optional
+	// AuthEncryptionKey - Selector to get the auth encryption key from the Secret
+	AuthEncryptionKey string `json:"authEncryptionKey"`
```

Comment:
this one is only used in heat, right. we should not have it in all the other services, where there is no usage of the parameter

Reply:
Nova operator uses `MetadataSecret` for similar purpose [1]. Selecting field from secret is pretty obvious extension of the selector that may be introduced in other operators at some point. And it would be much easier if the struct had support for it. 
On the other hand, one empty field won't tax resources in a noticeable way.

[1] https://github.com/openstack-k8s-operators/nova-operator/blob/9ea2124e10e9a748633e2edf5db3d7446f653be9/api/v1beta1/common_types.go#L79

Reply:
not sure I get your comment. you say AuthEncryptionKey is expected to come in other operators?

Reply:
I'm saying that it isn't there now, but that does make sense to have capability to select field from retrieved secret. The use case is general enough that I can see it showing up in other operators. And if it does not, which is possible, the field just won't be used.

Reply:
but we should not have fields in operators which are not being used. so I am not convinced a shared Passwordselector should have e.g. `authEncryptionKey`

Reply:
also we have to keep in mind that in future, when there is the need for new field it will be added to all the operators. we do not have a versioning for api structs, like it is for the apis in the lib-common. at least not right now.

Reply:
In that case the only option is to stick with a single selector ... it's not going to be as useful in the long run, and I'm not really sure it's that bad to have one extra field.

Reply:
es mentioned on the other parameter, you can also not default it to an per operator specific value.

Reply:
I'm not, there is no default there. Just `Optional`.

Reply:
for authEncryptionKey yes, which I think should not be in operators which don't use it,  but we'd need a default for Service

Reply:
Doing that with kubebuilder annotation is obviously not going to work, not when the struct is defined here. We can however do that with a defaulting webhook.

Reply:
yes, but wouldn't it be then much easier/simpler to keep the struct in the operator itself, instead of adding a defaulting logic?

Reply:
If you see that is a bridge too far than I see no reason to keep this PR open.